### PR TITLE
Crove App - Updates generate pdf from template action to use new api

### PR DESCRIPTION
**Pull request by croveapp**
### Changes with this update
- We are eliminating the need to use multiple apis by instead using single api to perform the required task.
- Adding background mode input prop.

documentation for api https://crove.stoplight.io/docs/crove-external-integrations-api/4c583ad42dad0-template-detail

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Review by vellames — CHANGES_REQUESTED** [review]
Hi there. Thanks for your contribution. Instead of using the`_makeRequest` method here. It would be great if you could create a method that calls `_makeRequest` inside the `app`.
If you want some reference you can check a component called `siteleaf`

**Comment by vellames** [comment]
Also, looks like you have some lint problem

**Comment by croveapp** [comment]
@vellames We have made the changes as requested. Kindly have a look.

File: components/crove_app/actions/generate-template-pdf/generate-template-pdf.mjs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      optional: true,
+    },
+  },
+  async additionalProps() {
+    let templateId = this.template_id;
+    let config = {
+      url: `${this.croveApp._getBaseUrl()}/templates/${templateId}/`,
+      method: "GET",
+    };
+    let resp = await this.croveApp._makeRequest(config);
```
Comment (vellames) [inline components/crove_app/actions/generate-template-pdf/generate-template-pdf.mjs:31]: Could you please create a method inside `app` and call the method here?

**Review by vellames — CHANGES_REQUESTED** [review]
Hey. Thanks for the changes! I missed one suggestion in `generate-template-pdf.mjs`, sorry about that. Once you do that, we are good to go

**Comment by croveapp** [comment]
@vellames the changes you requested have been done please have a look.

File: components/crove_app/actions/generate-template-pdf/generate-template-pdf.mjs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      label: "Background Mode",
+      description: "Whether to generate pdf in background mode or not.",
+      optional: true,
+    },
+  },
+  async additionalProps() {
+    let resp = await this.croveApp.getTemplateDetails(this.template_id);
+    let symbolTable = resp.symbol_table;
+    let props = {};
+    for (var k in symbolTable) {
```
Comment (feyzullah) [inline components/crove_app/actions/generate-template-pdf/generate-template-pdf.mjs:29]:
```suggestion
    for (const k in symbolTable) {
```

File: components/crove_app/actions/generate-template-pdf/generate-template-pdf.mjs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    return props;
+  },
+  async run() {
+
+    let resp = await this.croveApp.getTemplateDetails(this.template_id);
+
+    let symbolTable = resp.symbol_table;
+    let response = {};
+    for (var k in symbolTable) {
```
Comment (feyzullah) [inline components/crove_app/actions/generate-template-pdf/generate-template-pdf.mjs:44]:
```suggestion
    for (const k in symbolTable) {
```

**Review by feyzullah — COMMENTED** [review]
Hi @croveapp, this looks good to me, just changed `var`s with `const`s

**Review by feyzullah — APPROVED** [review]
Ready for QA

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[CroveApp_4893_2241.pdf](https://github.com/PipedreamHQ/pipedream/files/10209601/CroveApp_4893_2241.pdf)

**Pull request merged**